docs/tutorials/sinks/2_http_sink.md

```diff
@@ -0,0 +1,504 @@
+    stream::DriverResponse,
+    tls::TlsSettings,
+};
+```
+
+# Configuration
+
+First we want to update our config to allow an endpoint to be specified. Add this
+field to the `BasicConfig` struct:
+
+```rust
+    /// The endpoint to send HTTP traffic to.
+    #[configurable(metadata(
+        docs::examples = "http://localhost:3000/",
+        docs::examples = "http://example.com/endpoint/",
+    ))]
+    pub endpoint: String,
+```
+
+Every field in the configuration struct must have a comment. This is used to 
+generate documentation for the Sink. The metadata attribute added here is used
+to generate examples for the documentation. (This is possible because the config
+struct is annotated with `#[configurable_component(sink("basic"))]`).
+
+We then want to update our sink to take the endpoint from the config. At the same 
+time let's create an `HttpClient` that will handle sending the data. `HttpClient` 
+is our wrapper over [`hyper`][hyper_docs] used to send data over http.
+
+Update the `BasicSink` struct to look like: 
+
+```rust
+#[derive(Debug, Clone)]
+struct BasicSink {
+    endpoint: String,
+    client: HttpClient,
+}
+
+impl BasicSink {
+    pub fn new(config: &BasicConfig) -> Self {
+        let tls = TlsSettings::from_options(&None).unwrap();
+        let client = HttpClient::new(tls, &Default::default()).unwrap();
+        let endpoint = config.endpoint.clone();
+
+        Self { client, endpoint }
+    }
+}
+```
+
+# Encoder
+
+Now we want to create an Encoder that will take our event and convert it 
+to raw bytes.
+
+```rust
+#[derive(Clone)]
+struct BasicEncoder;
+```
+
+The Encoder must implement the `Encoder` trait:
+
+```rust
+impl Encoder<Event> for BasicEncoder {
+    fn encode_input(
+        &self,
+        input: Event,
+        writer: &mut dyn std::io::Write,
+    ) -> std::io::Result<usize> {
+    }
+}
+```
+
+The `Encoder` trait is generic over the type of input that we are 
+expecting. In our case it is `Event` since we will be encoding a single
+event at a time. Other Sinks may encode a `Vec<Event>`  if they are 
+sending batches of events, or they may send a completely different type
+if each event is processed in someway prior to encoding.
+
+
+The `encode_input` serializes the event to a String and writes these
+bytes:
+ 
+```rust
+    fn encode_input(
+        &self,
+        input: Event,
+        writer: &mut dyn std::io::Write,
+    ) -> std::io::Result<usize> {
+        let event = serde_json::to_string(&input).unwrap();
+        write_all(writer, 1, event.as_bytes()).map(|()| event.len())
+    }
+```
+
+# Request Builder
+
+Next we creata request builder that turns the event into a request. The 
+request is used by a Tower service that is responsible for actually sending
+the data.
+
+The request looks like:
+
+```rust
+#[derive(Clone)]
+struct BasicRequest {
+    payload: Bytes,
+    finalizers: EventFinalizers,
+    metadata: RequestMetadata,
+}
+```
+
+The fields in the request are:
```

Comment:
Are all of these fields necessary? I think they are.

```suggestion
The required fields in the request are:
```

Reply:
There's nothing that strictly requires them, it's just the data that our tower service will need in order to send the data out.. The fields are there because they are useful, not because they are mandatory.